compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt

```diff
@@ -158,11 +158,49 @@ internal abstract class ComposeBridge(
         }
     }
 
-    protected val sceneDimension: Dimension
-        get() = Dimension(
-            (scene.contentSize.width / component.density.density).toInt(),
-            (scene.contentSize.height / component.density.density).toInt()
+    /**
+     * Provides the maximum constraints where ComposeScene can be rendered.
+     * Dimensions' sum of all displays is used for that.
+     * Since we cannot show more than on all the screens.
+     *
+     * See [scenePreferredSize]
+     */
+    private fun getMaxConstraints(): Constraints {
+        val ge = GraphicsEnvironment.getLocalGraphicsEnvironment()
+        val deviceBounds = ge.screenDevices.map { it.defaultConfiguration.bounds }
+
+        val totalWidth = deviceBounds.sumOf { it.width }
+        val totalHeight = deviceBounds.sumOf { it.height }
+
+        val density = component.density.density
+        return Constraints(
+            maxWidth = (totalWidth * density).toInt(),
+            maxHeight = (totalHeight * density).toInt()
         )
+    }
+
+    /**
+     * Provides the size of ComposeScene content inside max possible constraints
+     *
+     * This is needed for the bridge between Compose and Swing since
+     * in some cases, Swing's LayoutManagers need
+     * to calculate the preferred size of the content without max/min constraints
+     * to properly lay it out.
+     *
+     * Example: Compose content inside Popup without a preferred size.
+     * Swing will calculate the preferred size of the Compose content and set Popup's side for that.
+     *
+     * See [androidx.compose.ui.awt.ComposePanelTest] test `initial panel size of LazyColumn with border layout`
+     */
+    protected val scenePreferredSize: Dimension
+        get() {
+            val contentSize =
+                scene.contentSizeInConstraints(getMaxConstraints())
```

Comment:
Comparing to the previous solution, this is much worse from performance side, since before we measured only in constraints no more than given size, and now we always measure in max constraints

Reply:
Just thought that we can use max(displays.dimenstion) here, not sum

Reply:
I would do this in a function, since it's not a simple/quick operation.

Reply:
Decided to proceed with Infinity constraints